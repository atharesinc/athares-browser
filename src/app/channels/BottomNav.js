import React from "react";
import { Link } from "react-router-dom";
import FeatherIcon from "feather-icons-react";

const BottomNav = ({ show, activeCircle }) => {
    // ask the user to log in
    if (!show) {
        return (
            <Link className="w-100" to={"/login"}>
                <div
                    className="w-100 v-mid bg-theme-dark flex flex-row justify-start items-center pv2 ph3"
                    id="bottom-nav"
                >
                    <FeatherIcon
                        icon={"log-in"}
                        className="white w2 h2 mr3"
                        style={{ height: "1.5em", width: "1.5em" }}
                    />
                    <div className="white">Login or Register</div>
                </div>
            </Link>
        );
    }
    // user is logged in but no circle is selected
    if (show && !activeCircle) {
        return (
            <div
                className="w-100 v-mid bg-theme-dark flex flex-row justify-start items-center pv2 ph3"
                                    id="bottom-nav"

            >
                <div className="white">Select a Circle</div>
            </div>
        );
    }
    // user is logged in and able to add a user to a defined circle
    return (
        <Link
            className="w-100"
            to={"/app/circle/" + activeCircle + "/add/user"}
        >
            <div
                className="w-100 v-mid bg-theme-dark flex flex-row justify-start items-center pv2 ph3"
                    id="bottom-nav"
            >
                <FeatherIcon
                    icon={"user-plus"}
                    className="white w2 h2 mr3"
                    style={{ height: "1.5em", width: "1.5em" }}
                />
                <div className="white">Invite User to Circle</div>
            </div>
        </Link>
    );
};
export default BottomNav;
