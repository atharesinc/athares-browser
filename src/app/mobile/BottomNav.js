import React, { Component } from "react";

import FeatherIcon from "feather-icons-react";

const BottomNav = props => (
  <div
    className="w-100 v-mid bg-theme-dark flex flex-row justify-start items-center pv2 ph3"
    style={{ height: "3em" }}
  >
    <FeatherIcon
      icon="plus"
      className="white w2 h2 mr3"
      style={{ height: "1.5em", width: "1.5em" }}
    />
    <div className="white">Invite User</div>
  </div>
);
export default BottomNav;
