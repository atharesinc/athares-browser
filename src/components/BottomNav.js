import React from 'reactn';
import { Link } from 'react-router-dom';
import { LogIn, UserPlus } from 'react-feather';

const BottomNav = ({ loggedIn, activeCircle, belongsToCircle }) => {
  // ask the user to log in
  if (!loggedIn) {
    return (
      <Link className='w-100' to={'/auth'}>
        <div
          className='w-100 v-mid bg-theme-dark flex flex-row justify-start items-center pv2 ph3'
          id='bottom-nav'
        >
          <LogIn className='white w2 h2 mr3' style={styles.icon} />
          <div className='white'>Login or Register</div>
        </div>
      </Link>
    );
  }

  // user is logged in but no circle is selected
  if (!activeCircle) {
    return (
      <div
        className='w-100 v-mid bg-theme-dark flex flex-row justify-start items-center pv2 ph3'
        id='bottom-nav-no-hover'
      >
        <div className='white-50'>Select a Circle</div>
      </div>
    );
  }

  // user is logged in and able to add a user to a defined circle
  if (belongsToCircle) {
    return (
      <Link className='w-100' to={'/app/circle/' + activeCircle + '/add/user'}>
        <div
          className='w-100 v-mid bg-theme-dark flex flex-row justify-start items-center pv2 ph3'
          id='bottom-nav'
        >
          <UserPlus className='white w2 h2 mr3' style={styles.icon} />
          <div className='white'>Invite User to Circle</div>
        </div>
      </Link>
    );
  }

  // user is logged in and viewing a valid circle, but doesn't belong to it
  return (
    <div
      className='w-100 v-mid bg-theme-dark flex flex-row justify-start items-center pv2 ph3'
      id='bottom-nav-no-hover'
    >
      <div className='white-50'> You don't belong to this Circle</div>
    </div>
  );
};

export default BottomNav;

const styles = {
  icon: { height: '1.5em', width: '1.5em' },
};
