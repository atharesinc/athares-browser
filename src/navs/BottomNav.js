import React from 'reactn';
import { Globe, Grid, Hash, User } from 'react-feather';
import { Link } from 'react-router-dom';

const BottomNav = ({ show, activeTab, toggleMenu }) => {
  return (
    <div id='bottom-nav' style={{ display: show ? 'none' : 'flex' }}>
      <Link className='bottom-nav-wrapper' to={'/app/circles'}>
        <Globe className={activeTab === 'circles' ? 'black' : 'black-60'} />
      </Link>
      <Link className='bottom-nav-wrapper' to={'/app/channels'}>
        <Hash className={activeTab === 'channels' ? 'black' : 'black-60'} />
      </Link>
      <Link className='bottom-nav-wrapper' to={'/app'}>
        <Grid className={activeTab === 'app' ? 'black' : 'black-60'} />
      </Link>
      <div className='bottom-nav-wrapper' onClick={toggleMenu}>
        <User className={activeTab === 'user' ? 'black' : 'black-60'} />
      </div>
    </div>
  );
};

export default BottomNav;
