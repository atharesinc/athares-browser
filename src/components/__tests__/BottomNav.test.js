import { create } from "react-test-renderer";
import BottomNav from "../BottomNav.js";
import React from "react";
import { BrowserRouter } from "react-router-dom";

describe("Bottom Nav component", () => {
  it("should render default bottomNav component", () => {
    const tree = create(
      <BrowserRouter>
        <BottomNav />
      </BrowserRouter>
    );

    expect(tree).toMatchSnapshot();
  });
  it("should render default bottomNav component when visible and user belongs to circle", () => {
    const tree = create(
      <BrowserRouter>
        <BottomNav show={true} belongsToCircle={true} />
      </BrowserRouter>
    );

    expect(tree).toMatchSnapshot();
  });
  it("should render default bottomNav component when show is false", () => {
    const tree = create(
      <BrowserRouter>
        <BottomNav show={false} />
      </BrowserRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
