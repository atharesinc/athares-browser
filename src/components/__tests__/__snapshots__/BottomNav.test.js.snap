// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bottom Nav component should render default bottomNav component 1`] = `
<a
  className="w-100"
  href="/login"
  onClick={[Function]}
>
  <div
    className="w-100 v-mid bg-theme-dark flex flex-row justify-start items-center pv2 ph3"
    id="bottom-nav"
  >
    <svg
      className="feather feather-log-in white w2 h2 mr3"
      fill="none"
      height={24}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "height": "1.5em",
          "width": "1.5em",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <g
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M15 3h4a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-4\\"></path><polyline points=\\"10 17 15 12 10 7\\"></polyline><line x1=\\"15\\" y1=\\"12\\" x2=\\"3\\" y2=\\"12\\"></line>",
          }
        }
      />
    </svg>
    <div
      className="white"
    >
      Login or Register
    </div>
  </div>
</a>
`;

exports[`Bottom Nav component should render default bottomNav component when show is false 1`] = `
<a
  className="w-100"
  href="/login"
  onClick={[Function]}
>
  <div
    className="w-100 v-mid bg-theme-dark flex flex-row justify-start items-center pv2 ph3"
    id="bottom-nav"
  >
    <svg
      className="feather feather-log-in white w2 h2 mr3"
      fill="none"
      height={24}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "height": "1.5em",
          "width": "1.5em",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <g
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M15 3h4a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-4\\"></path><polyline points=\\"10 17 15 12 10 7\\"></polyline><line x1=\\"15\\" y1=\\"12\\" x2=\\"3\\" y2=\\"12\\"></line>",
          }
        }
      />
    </svg>
    <div
      className="white"
    >
      Login or Register
    </div>
  </div>
</a>
`;

exports[`Bottom Nav component should render default bottomNav component when visible and user belongs to circle 1`] = `
<div
  className="w-100 v-mid bg-theme-dark flex flex-row justify-start items-center pv2 ph3"
  id="bottom-nav-no-hover"
>
  <div
    className="white-50"
  >
    Select a Circle
  </div>
</div>
`;
